cc1120_arduino/cc1120_spi_tests.c

```diff
@@ -153,65 +233,128 @@ bool cc1120_test_spi_strobe() {
+            return status;
         }
     }
 
-    if (status) {
+    if (status == CC1120_ERROR_CODE_SUCCESS) {
         uint8_t rxLastPos = 5;
         /* When the size of the RXFIFO is 0, the first byte must be ignored. See section 3.2.3 of datasheet. */
         uint8_t ignore;
         w_data = 0x0EU;
-        if (!cc1120_write_ext_addr_spi(CC1120_REGS_EXT_RXLAST, &rxLastPos, 1) ||
-                !cc1120_write_fifo_direct(CC1120_FIFO_RX_START, &w_data, 1) ||
-                !cc1120_read_fifo(&ignore, 1) ||
-                !cc1120_read_fifo(&r_data, 1)) {
+
+        status = cc1120_write_ext_addr_spi(CC1120_REGS_EXT_RXLAST, &rxLastPos, 1);
+
+        if (status != CC1120_ERROR_CODE_SUCCESS) {
+            mcu_log(CC1120_LOG_LEVEL_ERROR, "CC1120 FIFO test failed.\n");
+            return status;
+        } 
+
+        status = cc1120_write_fifo_direct(CC1120_FIFO_RX_START, &w_data, 1);
+
+        if (status != CC1120_ERROR_CODE_SUCCESS) {
+            mcu_log(CC1120_LOG_LEVEL_ERROR, "CC1120 FIFO test failed.\n");
+            return status;
+        } 
+
+        status = cc1120_read_fifo(&ignore, 1);
+
+        if (status != CC1120_ERROR_CODE_SUCCESS) {
+            mcu_log(CC1120_LOG_LEVEL_ERROR, "CC1120 FIFO test failed.\n");
+            return status;
+        } 
+
+        status = cc1120_read_fifo(&r_data, 1);
+
+        if (status != CC1120_ERROR_CODE_SUCCESS) {
             mcu_log(CC1120_LOG_LEVEL_ERROR, "CC1120 FIFO test failed.\n");
-            status = false;
-        } else if (r_data != 0x0EU) {
+            return status;
+        } 
+        
+        if (r_data != 0x0EU) {
             mcu_log(CC1120_LOG_LEVEL_ERROR, "CC1120 FIFO test failed.\n");
             mcu_log(CC1120_LOG_LEVEL_ERROR, "FIFO_RX_START read 0x%02X, expected 0x%02X\n", r_data, 0x0EU);
-            status = false;
+            status = CC1120_ERROR_CODE_TEST_FIFO_READ_WRITE_SINGLE_DIRECT_WRITE_FAILED; // what is happening here?
+            return status;
         }
     }
 
-    if (status) {
+    if (status == CC1120_ERROR_CODE_SUCCESS) {
         uint8_t burstWriteData2[3] = {0x0FU, 0x10U, 0x11U};
         uint8_t burstReadData2[3];
-        if (!cc1120_write_fifo_direct(CC1120_FIFO_RX_START+1, burstWriteData2, 3) ||
-            !cc1120_read_fifo(burstReadData2, 3) ||
-            memcmp(burstWriteData2, burstReadData2, 3) != 0) {
+        bool incorrectValues = false;
+        
+        status = cc1120_write_fifo_direct(CC1120_FIFO_RX_START+1, burstWriteData2, 3);
+
+        if (status != CC1120_ERROR_CODE_SUCCESS) {
+            mcu_log(CC1120_LOG_LEVEL_ERROR, "CC1120 FIFO test failed.\n");
+            return status; 
+        }
+
+        status = cc1120_read_fifo(burstReadData2, 3);
+
+        if (status != CC1120_ERROR_CODE_SUCCESS) {
+            mcu_log(CC1120_LOG_LEVEL_ERROR, "CC1120 FIFO test failed.\n");
+            return status; 
+        }
+
+        if (memcmp(burstWriteData2, burstReadData2, 3)); {
```

Comment:
Return value of 0 indicates the strings are the same, so this should be more like `if (memcmp(burstWriteData2, burstReadData2, 3) != 0)`

Reply:
Fixed this in a recent commit, will resolve comment once you see this